tests/commands/canary.py

```diff
@@ -16,3 +16,11 @@ def test_cmd_canary(self):
         res = gdb_start_silent_cmd("canary", target=_target("canary"))
         self.assertNoException(res)
         self.assertIn("The canary of process", res)
+        # On other platforms, we read the canary from the auxiliary vector -
+        # overwriting this value does not overwrite the in-use canary so it
+        # is not tested
+        if platform.machine() == "x86_64":
```

Comment:
It's just cleaner to add another test with the decorator `@pytest.mark.skipif(ARCH != "x86_64", reason=f"Skipped for {ARCH}")`. That also makes the comment not necessary.